Oleksii_Fedorchuk/HW18_Oleksii_Fedorchuk/HW18_Oleksii_Fedorchuk/test_human.py

```diff
@@ -0,0 +1,102 @@
+import pytest
+
+
+@pytest.mark.parametrize("name, age, gender", [("Harry", 10, "male")], ids=["Human is created"])
+def test_human_creating(human_template, name, age, gender):
+    """
+    Pre-Condition:
+    create_human texture is created
+    Steps:
+    1. Add name
+    2. Add age
+    3. Add gender
+    ER: [Human is created]
+    """
+    human = human_template(name, age, gender)
+    assert human
+
+
+@pytest.mark.xfail
+@pytest.mark.parametrize("name, age, gender", [("Harry", -10, "male")])
+def test_human_negative_age(human_template, name, age, gender):
+    """
+    Pre-Condition:
+    create_human texture is created
+    Steps:
+    1. Add name
+    2. Add incorrect age
+    3. Add gender
+    ER: Human age test should be fell, the age cannot be negative
+    """
+    human = human_template(name, age, gender)
+    assert human.age >= 0, f"Human age cannot be less then 0"
+
+
+@pytest.mark.parametrize("name, age, gender", [("Harry", 12, "male"), ("Hermione", 11, "female")])
+def test_human_gender(human_template, name, age, gender):
+    """
+    Pre-Condition:
+    create_human texture is created
+    Steps:
+    1. Add name
+    2. Add age
+    3. Add gender
+    ER: Human should have gender
+    """
+    human = human_template(name, age, gender)
+    assert human.gender == gender
+
+
+def test_human_change_gender(human_template):
+    """
+    Pre-Condition:
+    create_human texture is created
+    Steps:
+    1. Add name
+    2. Add age
+    3. Add gender
```

Comment:
These 3 steps can be in the precondition.
Pre-Condition: Human with valid gender created